### PR TITLE
[Main] Fix rest query

### PR DESCRIPTION
**Pull request by colmazia**
Fixed: Rest query for cosmos 0.46.x

## Implementation details

- used grpc instead of rest
- created new proto file for the grpc
- moved proof out of oracle module
- change go_package of proto to be aligned with go.mod (chain/v2)

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by RogerKSI — CHANGES_REQUESTED** [review]

**Review by RogerKSI — APPROVED** [review]

**Review by satawatnack — APPROVED** [review]

**Review by taobun — CHANGES_REQUESTED** [review]

File: proto/bandchain/v1/oracle/proof.proto [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+message SingleProof {
+  uint64 block_height = 1;
+  OracleDataProof oracle_data_proof = 2;
+  BlockRelayProof block_relay_proof = 3;
+}
+
+// MultiProof contains block height, list of oracle data proof and block relay proof
+message MultiProof {
+  uint64 block_height = 1;
+  repeated OracleDataProof oracle_data_multi_proof = 2;
```
Comment (taobun) [inline proto/bandchain/v1/oracle/proof.proto:93]: Please add `[ (gogoproto.nullable) = false ]` to generate `[]OracleDataProof` not `[]*OracleDataProof`

File: proto/bandchain/v1/oracle/proof.proto [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  uint64 block_height = 1;
+  RequestsCountProof count_proof = 2;
+  BlockRelayProof block_relay_proof = 3;
+}
+
+// OracleDataProof contains result, version and merkle paths
+message OracleDataProof {
+  oracle.v1.Result result = 1;
+  uint64 version = 2;
+  repeated IAVLMerklePath merkle_paths = 3;
```
Comment (taobun) [inline proto/bandchain/v1/oracle/proof.proto:108]: ditto

File: proto/bandchain/v1/oracle/proof.proto [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  uint64 subtree_version = 4;
+  bytes sibling_hash = 5;
+}
+
+// BlockRelayProof contains multi store proof, block header merkle parts, common encoded part and signatures
+message BlockRelayProof {
+  MultiStoreProof multi_store_proof = 1;
+  BlockHeaderMerkleParts block_header_merkle_parts = 2;
+  CommonEncodedVotePart common_encoded_vote_part = 3;
+  repeated TMSignature signatures = 4;
```
Comment (taobun) [inline proto/bandchain/v1/oracle/proof.proto:125]: ditto

**Review by taobun — CHANGES_REQUESTED** [review]

File: client/grpc/oracle/proof/proof.pb.go [hunk 4]
```diff
@@ -0,0 +1,5685 @@
...
+
+func (m *BlockHeaderMerkleParts) GetEvidenceAndProposerHash() []byte {
+	if m != nil {
+		return m.EvidenceAndProposerHash
+	}
+	return nil
+}
+
+// CommonEncodedVotePart represents the common part of encoded vote
+type CommonEncodedVotePart struct {
```
Comment (taobun) [inline client/grpc/oracle/proof/proof.pb.go:1072]: We used `hexBytes` to make json encode as hex if you use `[]byte` it will encode with base 64

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by RogerKSI — APPROVED** [review]

File: client/grpc/node/service.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+// RegisterNodeService registers the node gRPC service on the provided gRPC router.
+func RegisterNodeService(clientCtx client.Context, server gogogrpc.Server) {
+	RegisterServiceServer(server, NewQueryServer(clientCtx))
+}
+
+// RegisterGRPCGatewayRoutes mounts the node gRPC service's GRPC-gateway routes
+// on the given mux object.
+func RegisterGRPCGatewayRoutes(clientConn gogogrpc.ClientConn, mux *runtime.ServeMux) {
+	_ = RegisterServiceHandlerClient(context.Background(), mux, NewServiceClient(clientConn))
```
Comment (taobun) [inline client/grpc/node/service.go:24]: Just remove `_ = `

File: client/grpc/node/service.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
+package node
+
+import (
+	context "context"
```
Comment (taobun) [inline client/grpc/node/service.go:4]: No need to rename

File: client/grpc/node/service.go (hunk 5 above)
Comment (RogerKSI) [reply client/grpc/node/service.go]: removed

File: client/grpc/node/service.go (hunk 6 above)
Comment (RogerKSI) [reply client/grpc/node/service.go]: removed

**Review by taobun — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**